### DISCUSSION TITLE
documentation on .buildpacks file

### DISCUSSION THREAD

**Discussion by shijialee** [dokku/dokku#7307] [Q&A] [answered]
Hi,

Just started playing around with dokku. I am using buildpacks and the repo I am testing with has a Dockerfiles at the root directory. To avoid automatic Dockerfile deployment detection, I created an empty `.buildpacks` file at the root directory per the documentation https://dokku.com/docs/deployment/builders/herokuish-buildpacks/ 

But I couldn't deploy with `git push dokku myapp` and I guess it had issue detecting my app. 

```
-----> Set test-deploy as deploy-branch
-----> Cleaning up...
-----> Building douapp from herokuish
-----> Adding BUILD_ENV to build environment...
       BUILD_ENV added successfully
-----> Warning: Multiple default buildpacks reported the ability to handle this app. The first buildpack in the list below will be used.
       Detected buildpacks: multi python
-----> Multipack app detected
-----> Discovering process types
       Procfile declares types -> web
-----> Releasing douapp...
-----> Checking for predeploy task
       No predeploy task found, skipping
-----> Checking for release task
       No release task found, skipping
-----> Checking for first deploy postdeploy task
       No first deploy postdeploy task found, skipping
=====> Processing deployment checks
```

I found the following old documentation on `.buildpack` at https://dokku.com/docs~v0.6.5/deployment/methods/buildpacks/

_You can only set a single buildpack using the BUILDPACK_URL, though there may be times when you wish to use multiple buildpacks. To do so, simply create a .buildpacks file in the base of your repository. This file should list all the buildpacks, one-per-line. For instance, if you wish to use both the nodejs and ruby buildpacks, your .buildpacks file should contain the following:_ 

```
https://github.com/heroku/heroku-buildpack-nodejs.git#v87
https://github.com/heroku/heroku-buildpack-ruby.git#v142
```

After adding `https://github.com/heroku/heroku-buildpack-python.git` the deploy worked, I am still getting multiple default buildpacks warning though, I thought the `.buildpack` file should have fixed that. 

```
-----> Cleaning up...
-----> Building douapp from herokuish
-----> Adding BUILD_ENV to build environment...
       BUILD_ENV added successfully
-----> Warning: Multiple default buildpacks reported the ability to handle this app. The first buildpack in the list below will be used.
       Detected buildpacks: multi python
-----> Multipack app detected
=====> Downloading Buildpack: https://github.com/heroku/heroku-buildpack-python
=====> Detected Framework: Python
```

I could set the python buildpack via ENV but I wish there were more documentation about the `.buildpack` file. 

Am I doing it correctly and how do I get rid of the warning?

Thanks

**Comment by josegonzalez** [comment, marked as answer]
You can force-set a builder by doing:

```shell
dokku builder:set node-js-app selected herokuish
```

There is definitely a use case for an appendix that documents all the file formats we have. Mind filing a ticket for that?

**Reply by shijialee** [reply]
Thanks. I created one here https://github.com/dokku/dokku/issues/7315
